src/pages/home/sidebar/SidebarLinksData.js

```diff
@@ -0,0 +1,177 @@
+import React, {useMemo, useRef} from 'react';
+import _ from 'underscore';
+import {deepEqual} from 'fast-equals';
+import {withOnyx} from 'react-native-onyx';
+import PropTypes from 'prop-types';
+import lodashGet from 'lodash/get';
+import SidebarUtils from '../../../libs/SidebarUtils';
+import SidebarLinks, {basePropTypes} from './SidebarLinks';
+import withCurrentReportID from '../../../components/withCurrentReportID';
+import compose from '../../../libs/compose';
+import ONYXKEYS from '../../../ONYXKEYS';
+import reportPropTypes from '../../reportPropTypes';
+import CONST from '../../../CONST';
+
+const propTypes = {
+    ...basePropTypes,
+
+    /* Onyx Props */
+    /** List of reports */
+    chatReports: PropTypes.objectOf(reportPropTypes),
+
+    /** All report actions for all reports */
+    allReportActions: PropTypes.objectOf(
+        PropTypes.arrayOf(
+            PropTypes.shape({
+                error: PropTypes.string,
+                message: PropTypes.arrayOf(
+                    PropTypes.shape({
+                        moderationDecisions: PropTypes.arrayOf(
+                            PropTypes.shape({
+                                decision: PropTypes.string,
+                            }),
+                        ),
+                    }),
+                ),
+            }),
+        ),
+    ),
+
+    /** Whether the personal details are loading. When false it means they are ready to be used. */
+    isPersonalDetailsLoading: PropTypes.bool,
+
+    /** The chat priority mode */
+    priorityMode: PropTypes.string,
+
+    /** Beta features list */
+    betas: PropTypes.arrayOf(PropTypes.string),
+
+    /** The policies which the user has access to */
+    // eslint-disable-next-line react/forbid-prop-types
+    policies: PropTypes.object,
+};
+
+const defaultProps = {
+    chatReports: {},
+    allReportActions: {},
+    priorityMode: CONST.PRIORITY_MODE.DEFAULT,
+    isPersonalDetailsLoading: true,
+    betas: [],
+    policies: [],
+};
+
+function SidebarLinksData({allReportActions, betas, chatReports, currentReportID, insets, isPersonalDetailsLoading, isSmallScreenWidth, onLayout, onLinkClick, policies, priorityMode}) {
+    const reportIDsRef = useRef([]);
+    const optionListItems = useMemo(() => {
+        const reportIDs = SidebarUtils.getOrderedReportIDs(currentReportID, chatReports, betas, policies, priorityMode, allReportActions);
+        if (deepEqual(reportIDsRef.current, reportIDs)) {
+            return reportIDsRef.current;
+        }
+        reportIDsRef.current = reportIDs;
```

Comment:
That's because even when we put the same values to `SidebarUtils.getOrderedReportIDs` it will always return a new array reference. The same goes for every selector. So it can happen that one of the prop updates, but the real content of it stayed the same. In this case the `useMemo` will run and re-calculate the list of report IDs. The list will be exactly the same, but it's a new ref.If we'd return the new ref it would cause a re-render of `SidebarLinks`.
And we always must return from the `useMemo` hook. That's why we store the results in a ref, so in case its the very same result we can return the same reference.